var/spack/repos/builtin/packages/eigen/package.py

```diff
@@ -80,8 +86,20 @@ def cmake_args(self):
             # CMake fails without this flag
             # https://gitlab.com/libeigen/eigen/-/issues/1656
             args += [self.define("BUILD_TESTING", "ON")]
+        if self.spec.satisfies("+rocm"):
+            args.append(self.define("ROCM_PATH", self.spec["hip"].prefix))
+            args.append(self.define("HIP_PATH", self.spec["hip"].prefix))
+            args.append(self.define("EIGEN_TEST_HIP", "ON"))
+            args.append(self.define("Boost_INCLUDE_DIR", self.spec["boost"].prefix.include))
```

Comment:
... unless I misunderstood and the `rocm` portion has a hard dependency on `boost`?

Reply:
No this is my mistake, this should only be present when building the test

Reply:
Almost there, this should be `@master+test` right? By default`@master` won't have the right directory..

Reply:
yes I've added the condition `self.run_tests`

Reply:
Sorry, actually I'm still a bit hazy on this. Right now it seems like `boost` is needed if `tests` are run with `master`. Is this true? AFAIK shouldn't it be the `nightly` tests or if `rocm` is needed then the testsuite with `master` will need `boost`?

Reply:
No worries, from what I see here: https://gitlab.com/libeigen/eigen/-/blob/master/test/CMakeLists.txt#L385, there is a find_package call for boost for tests, but it's not a hard dependency.